engine/src/main/java/org/pentaho/di/trans/steps/systemdata/SystemDataMetaInjectionTypeConverter.java

```diff
@@ -0,0 +1,47 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2002-2018 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.trans.steps.systemdata;
+
+import org.pentaho.di.core.exception.KettleValueException;
+import org.pentaho.di.core.injection.InjectionTypeConverter;
+
+public class SystemDataMetaInjectionTypeConverter extends InjectionTypeConverter {
+
+  @Override
+  public Enum<?> string2enum( Class<?> enumClass, String v ) throws KettleValueException {
+    if ( v == null ) {
+      return null;
+    }
+
+    // For SystemDataMeta, enum should be a SystemDataTypes
+    SystemDataTypes type = SystemDataTypes.getTypeFromString( v );
+    if ( !v.equals( SystemDataTypes.TYPE_SYSTEM_INFO_NONE.toString() ) && type == SystemDataTypes.TYPE_SYSTEM_INFO_NONE ) {
```

Comment:
fyi, switching the order of the equals check saves you a 'v' null check: !SystemDataTypes.TYPE_SYSTEM_INFO_NONE.toString().equals( v )

Also, why not just simply call SystemDataTypes.getTypeFromString( v )? Why return null if v is null, throw an exception when type is 'none' (but only when value isn't explicitly set to 'none') and return the enum in other cases?

Reply:
Specifically here https://github.com/pentaho/pentaho-kettle/blob/master/engine/src/test/java/org/pentaho/di/core/injection/BaseMetadataInjectionTest.java#L145-L146 it checks for a bad enum to see it fail. Perhaps I should just have the getTypeFromString( v ) throw the exception, I was just trying to get around having all calls to that function to have to be wrapped in a try/catch.